File: custom-addons/approvals/models/mail_activity.py
# -*- coding: utf-8 -*-
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models


class MailActivity(models.Model):
    _inherit = 'mail.activity'

    def activity_format(self):
        result = super(MailActivity, self).activity_format()
        activity_type_approval_id = self.env.ref('approvals.mail_activity_data_approval').id
        for activity in result:
            if activity['activity_type_id'] and activity['activity_type_id'][0] == activity_type_approval_id and \
                    activity['res_model'] == 'approval.request':
                request = self.env['approval.request'].browse(activity['res_id'])
                approver = request.approver_ids.filtered(lambda approver: activity['user_id'][0] == approver.user_id.id)
                activity['approver_id'] = approver.id
                activity['approver_status'] = approver.status
        return result

File: custom-addons/approvals/static/tests/helpers/mock_server/models/mail_activity.js
/** @odoo-module **/

// ensure mail override is applied first.
import '@mail/../tests/helpers/mock_server/models/mail_activity';

import { patch } from "@web/core/utils/patch";
import { MockServer } from '@web/../tests/helpers/mock_server';

patch(MockServer.prototype, {
    /**
     * @override
     */
    _mockMailActivityActivityFormat(ids) {
        const activities = super._mockMailActivityActivityFormat(ids);
        for (const activity of activities) {
            if (activity.res_model === 'approval.request') {
                // check on activity type being approval not done here for simplicity
                const approver = this.getRecords('approval.approver', [
                    ['request_id', '=', activity.res_id],
                    ['user_id', '=', activity.user_id[0]],
                ])[0];
                if (approver) {
                    activity.approver_id = approver.id;
                    activity.approver_status = approver.status;
                }
            }
        }
        return activities;
    },
});

